client/state/data-layer/wpcom/sites/activity/from-api.js

```diff
@@ -28,6 +28,55 @@ export function transformer( apiResponse ) {
 	return map( orderedItems, processItem );
 }
 
+const getActivityDescription = item => {
+	const { generator, object } = item;
+
+	switch ( item.name ) {
+		case 'post__published':
+			return (
+				`<a href="/read/blogs/${ generator.blog_id }/posts/` +
+				`${ object.object_id }">${ object.name }</a>` +
+				'<div class="activity-log-item__event">Post published</div>'
+			);
+
+		case 'plugin__deleted':
+			return (
+				`<a href="/plugins/${ object.name }/${ generator.blog_id }">${ object.name }</a>` +
+				'<div class="activity-log-item__event">Plugin deleted</div>'
+			);
+
+		case 'plugin__installed':
+			return (
+				`<a href="/plugins/${ object.name }/${ generator.blog_id }">${ object.name }</a> (${ object.object_version })` +
+				'<div class="activity-log-item__event">Plugin installed</div>'
+			);
+
+		case 'user__deleted':
+			return (
+				`<a href="/people/edit/${ generator.blog_id }/${ object.name }">` +
+				`${ object.name }</a>` +
+				'<div class="activity-log-item__event">User removed</div>'
+			);
+
+		case 'user__registered':
+			return (
+				`<a href="/people/edit/${ generator.blog_id }/${ object.name }">` +
+				`${ object.name }</a>` +
+				'<div class="activity-log-item__event">User added</div>'
+			);
+
+		case 'user__updated':
```

Comment:
I'm not sure what this one does, but I guessed it was for when a user's role changed. If that's not correct, I can change the copy. Also, I hard-coded "Administrator" because I don't know how to pull in the relevant data.

Reply:
the data would have to come from the activity data in `item` - it's what we have available. otherwise we're getting into complex territory here that will need to happen in the API when we get there (take user id and then ask, what is this user's role).

in general I'd recommend _not_ pushing out code we don't understand; we're likely to embarrass ourselves. maybe we can find out what `user__udpated` means and then update here?